README
=== Plugin Name ===
Contributors: andrew.alba
Donate link: http://www.wordpress-plug.in/
Tags: jquery, notification, twitter
Requires at least: 1.3.2
Tested up to: 1.7+
Stable tag: 3.0

jQuery notification method similar to one found on Twitter site.

== Description ==

This jQuery plug-in was inspired by a similar effect you can see on the Twitter site. Some of the advantages: 

*   Notification toolbar displayed at the top or bottom of page (no scrolling to view error)
*	Multiple messages can be displayed in notification bar
*	Options to automatically close the notification bar after set delay
*	Notification bar closes with blur/'esc' key/'close' link

== Installation ==

This section describes how to install the plugin and get it working.

1.  Download jquery.showmessage and unzip/unpack on in your public directory
1.  Insert `<link type="text/css" href="/path/to/assets/css/showMessage.css" rel="stylesheet" screen="media"/> in the head of your document.
1.	Insert `<script type="text/javascript" src="/path/to/assets/js/jquery.showMessage.min.js" charset="utf-8"></script>` after jQuery is loaded (end of page).
1.	Call the plugin (default appends to <body/>. `$.showMessage({'thisMessage':['This is message one!','This is second message']});`

== Frequently Asked Questions ==

== Screenshots ==

== Changelog ==

= v3.0 =
* Complete rewrite to better harness jQuery methods and create API for ease of use.
* Replaced position option to reflect CSS style position attribute.
* Former position attribute name changed to location. Options remain ['top','bottom'].

= v2.4 =
* Found warning for unscoped variable 'event'. Changed line 60 to read keycode = window.event.keyCode;
* Added some different styles to the notification windows.

= v2.3 =
* Plug-in was overwriting the selected element with 'body' element (was accidently hard coded and never properly removed).
* Added another option for z-index. Ran into an issue where notification was being placed behind a light box. The new option will give more flexibility to and allow the plug-in to determine various z-index.

= v2.2 =
* Minor change as per recommendation by Sebastian Kun (http://www.arcestra.com/).
* Added options for better 'internationalization' of navigation text. Sebastian added two more options for closeLink text and closeText text.
* Removed the jquery.showMessage.pack.js option

= v2.1 =
* Minor change to option.position. Changed order so looking for top position else set to bottom.
* Issue with IE7+ if there is no doctype declared (shame on you, since DOCTYPES are vital to the proper functioning of web standards in browsers), we have changed insertion of #showMessage to either .prepend() or .append() depending on if position is top or bottom.

= v2.0 =
* Received a request to make some improvements to the plugin from DaveG. Thank you much for the constructive comments DaveG.
* Added new option useEsc to allow the 'esc' key to be optional (boolean true|false).
* Changed the global variable t to be included in the function and renamed to showMessage_t.
* Extended the plugin to use the jQuery.fn (does not return a value so this may not be chainable).

== Arbitrary section ==


src/CHANGELOG.TXT
2.4    :::::::::::....... 2011-04-18 .......::::::::::::::
// ********************************************************
	Found warning for unscoped variable 'event'. Changed 
	line 60 to read keycode = window.event.keyCode;
	
	Added some different styles to the notification
	windows.
	
2.3    :::::::::::....... 2010-12-20 .......::::::::::::::
// ********************************************************
	Plug-in was overwriting the selected element with
	'body' element (was accidently hard coded and never
	properly removed).
	
	Added another option for z-index. Ran into an issue
	where notification was being placed behind a light box.
	The new option will give more flexibility to and allow
	the plug-in to determine various z-index.

2.2    :::::::::::....... 2010-11-23 .......::::::::::::::
// ********************************************************
	Minor change as per recommendation by Sebastian Kun
	(http://www.arcestra.com/).
	
	Added options for better 'internationalization' of 
	navigation text. Sebastian added two more options for
	closeLink text and closeText text.
	
	Removed the jquery.showMessage.pack.js option

2.1    :::::::::::....... 2010-05-06 .......::::::::::::::
// ********************************************************
	Minor change to option.position. Changed order so
	looking for top position else set to bottom.
	
	Issue with IE7+ if there is no doctype declared (shame
	on you, since DOCTYPES are vital to the proper
	functioning of web standards in browsers), we have
	changed insertion of #showMessage to either .prepend()
	or .append() depending on if position is top or 
	bottom.
	
2.0    :::::::::::....... 2010-04-05 .......::::::::::::::
// ********************************************************
	Received a request to make some improvements to the 
	plugin from DaveG. Thank you much for the constructive
	comments DaveG.
	
	Added new option useEsc to allow the 'esc' key to be
	optional (boolean true|false).
	
	Changed the global variable t to be included in the
	function and renamed to showMessage_t.
	
	Extended the plugin to use the jQuery.fn (does not
	return a value so this may not be chainable).

returnForm.php
<?php
/*
 * returnForm.php
 * Alba Web Studio
 * http://albawebstudio.com/
 *
 * Copyright (c) 2010 Andrew Alba
 * Dual licensed under the MIT and GPL licenses.
 * http://docs.albawebstudio.com/License
 *
 * Date: Jan 31, 2010 12:06:50 PM
 * Project: dingobytes.com
 * Revision: 
 */
header('Content-Type: text/xml');
$error="true";
if( !isset($_GET['fullName']) )
{
	$returnMessage="Please enter your full name";
}
if( !isset($_GET['email']) || (isset($_GET['email']) && !eregi("^([a-zA-Z0-9_\.\-])+\@(([a-zA-Z0-9\-])+\.)+([a-zA-Z0-9]{2,4})+$", $_GET['email']) ) )
{
	$returnMessage="Please enter a valid email address";
}
if( isset($_GET['phone']) && !eregi("^(([0-9]{1})*[- .(]*([0-9a-zA-Z]{3})*[- .)]*[0-9a-zA-Z]{3}[- .]*[0-9a-zA-Z]{4})+$", $_GET['phone']) )
{
	$returnMessge="Please enter a valid phone number";
}
if( !isset($_GET['message']) )
{
	$returnMessage="Please enter a message before sending";
}
if( !isset($returnMessage) )
{
	$error="false";
	$returnMessage="Your message was sent successfully!";
}
?>
<response>
	<error><?php echo $error; ?></error>
	<returnMessage><?php echo $returnMessage; ?></returnMessage>
</response>

src/assets/js/jquery.showMessage.js
/**
 * @preserve showMessage - jQuery plugin
 * Simple Twitter like notification method
 *
 * Examples and documentation can be found at http://showMessage.dingobytes.com/
 *
 * Copyright (c) 2009 - 2012 Andrew Alba (http://albawebstudio.com)
 *
 * Version: 3.0
 * Requires jQuery v1.7+
 *
 * Dual licensed under the MIT  and GPL licenses:
 * 	 http://www.opensource.org/licenses/mit-license.php
 *   http://www.gnu.org/licenses/gpl.html
 *
 * Date: Tue Apr 04 18:00:00 2010 -0500
 */

/*global jQuery setTimeout window parent document event clearTimeout */
;(function($) {
	var showMessage_t, messageHolder, stateHolder, showmessageul, messageNav;

	/* Private methods */
	_autoClose = function(delayTime) {
		if (typeof(showMessage_t) != 'undefined') {
			clearTimeout(showMessage_t);
		}
		showMessage_t = setTimeout(function() {
			$('#showMessage', window.parent.document).fadeOut();
		}, delayTime);
	},

	_useEsc = function() {
		$(window).keydown(function(e) {
			var keycode;
			if (e === null) { // ie
				keycode = event.keyCode;
			} else { // mozilla
				keycode = e.which;
			}
			if (keycode == 27) { // close
				$('#showMessage', window.parent.document).fadeOut();
				if (typeof(showMessage_t) != 'undefined') {
					clearTimeout(showMessage_t);
				}
			}
		});
	},

	_navigation = function(useEsc, escText, closeText) {
		var messageNavigation = $('<span></span>').addClass('messageNav');
		if (useEsc) {
			$(messageNavigation).html(escText + ' ');
		}
		var closeText = $('<a></a>').attr({
			href: '',
			title: option.closeText
		}).css('text-decoration', 'underline').click(function() {
			$('#showMessage', window.parent.document).fadeOut();
			clearTimeout(showMessage_t);
			return false;
		}).text(closeText);
		$(messageNavigation).append(closeText);
		return messageNavigation;
	},

	_abandon = function() {
		$(window).click(function() {
			if ($('#showMessage', window.parent.document).length) {
				$('#showMessage', window.parent.document).fadeOut();
				$(window).unbind('click');
				if (typeof(showMessage_t) != 'undefined') {
					clearTimeout(showMessage_t);
				}
				return false;
			}
		});
	},

	_show = function(that, option) {
		// first clear all ui=widget
		try{
			if ($('#showMessage', window.parent.document).length) {
				$('#showMessage', window.parent.document).remove();
			}
			// create an messageHolder div
			messageHolder = $('<div></div>').css({
				position: option.position,
				'z-index': option.zIndex,
				filter: 'Alpha(Opacity=' + option.opacity * 100 + ')',
				opacity: option.opacity
			}).attr('id', 'showMessage').addClass(option.className);
			if (option.location == 'top') {
				$(messageHolder).css('top', 0);
			} else {
				$(messageHolder).css('bottom', 0);
			}
			if (option.useEsc) {
				_useEsc();
			} else {
				$(window).unbind('keydown');
			}
			if (option.displayNavigation) {
				messageNav = _navigation(option.useEsc, option.escText, option.closeText);
				$(messageHolder).append(messageNav);
			} else {
				_abandon();
			}

			showmessageul = $('<ul></ul>');

			// build messages from array
			for (var i = 0; i < option.thisMessage.length; i++) {
				var showmessageli = $('<li></li>').html(option.thisMessage[i]);
				$(showmessageul).append(showmessageli);
			}
			stateHolder = $('<div></div>').addClass('stateHolder').append(showmessageul);
			$(messageHolder).append(stateHolder);
			if (option.location == 'top') {
				$(that, window.parent.document).prepend(messageHolder);
			} else {
				$(that, window.parent.document).append(messageHolder);
			}
			$(messageHolder).fadeIn();
			if (option.autoClose) {
				_autoClose(option.delayTime);
			}
		} catch(e) {
			console.log('error message:', e.message);
		}
	},

	/* Public methods */
	$.fn.showMessage = function(options) {
		if (!$(this).length) {
			return this;
		}

		option = $.extend($.fn.showMessage.defaults, options);
		_show(this, option);
	};


	$.showMessage = function(obj) {
		option = $.extend($.fn.showMessage.defaults, obj);
		_show(jQuery('body', window.parent.document), option);
  	};

  	$.showMessage.close = function() {
		if ($('#showMessage', window.parent.document).length) {
			if (typeof(showMessage_t) != 'undefined') {
				clearTimeout(showmessage_t);
			}
			$('#showMessage', window.parent.document).fadeOut();
		}
		return false;
	};

	$.showMessage.init = function() {
		if ($("#showMessage").length) {
			return;
		}

	};

	//DEFAULT CONFIGURATION PROPERTIES
	$.fn.showMessage.defaults = {
		thisMessage: [''],
		className: 'notification',

		position: 'fixed',
		zIndex: 1001,
		opacity: .9,
		location: 'top',

		useEsc: true,
		displayNavigation: true,
		closeText: 'close',
		escText: 'Esc Key or',

		autoClose: false,
		delayTime: 5000,

		onStart : function(){},
		onCancel : function(){},
		onComplete : function(){},
		onCleanup : function(){},
		onClosed : function(){},
		onError : function(){}
	};

  	$(document).ready(function() {
		$.showMessage.init();
	});
})(jQuery);